protocol-units/settlement/mcr/manager/src/manager.rs

```diff
@@ -0,0 +1,182 @@
+use crate::{BlockCommitmentEvent, CommitmentEventStream, McrSettlementManagerOperations};
+
+use mcr_settlement_client::McrSettlementClientOperations;
+use movement_types::{BlockCommitment, BlockCommitmentRejectionReason};
+
+use async_stream::stream;
+use async_trait::async_trait;
+use tokio::sync::mpsc;
+use tokio_stream::StreamExt;
+
+use std::collections::BTreeMap;
+use std::mem;
+
+/// Public handle for the MCR settlement manager.
+pub struct Manager {
+	sender: mpsc::Sender<BlockCommitment>,
+}
+
+impl Manager {
+	/// Creates a new MCR settlement manager.
+	///
+	/// Returns the handle with the public API and the stream to receive commitment events.
+	/// The stream needs to be polled to drive the MCR settlement client and
+	/// process the commitments.
+	pub fn new<C: McrSettlementClientOperations + Send + 'static>(
+		client: C,
+	) -> (Self, CommitmentEventStream) {
+		let (sender, receiver) = mpsc::channel(16);
+		let event_stream = process_commitments(receiver, client);
+		(Self { sender }, event_stream)
+	}
+}
+
+#[async_trait]
+impl McrSettlementManagerOperations for Manager {
+	async fn post_block_commitment(
+		&self,
+		block_commitment: BlockCommitment,
+	) -> Result<(), anyhow::Error> {
+		self.sender.send(block_commitment).await?;
+		Ok(())
+	}
+}
+
+fn process_commitments<C: McrSettlementClientOperations + Send + 'static>(
```

Comment:
I'd really love to see a unit test or two, just testing `process_comitments`

Reply:
I believe this is exercised by the two unit tests I did add, that exercise the module's API. I'm not a fan of implementation detail tests, but if there's a problem with coverage (and it there is a good reason why the code can't be reached from public API), we can test the implementation fn as well.

Reply:
I can add tests in the same pattern to see if back-pressure works as expected, that will check more dynamic conditions.

Reply:
Yes more back pressure / load.  The unit tests here are good, but perhaps one more test a little more dynamic would be appreciated.

Reply:
Done in https://github.com/movementlabsxyz/movement/pull/58/commits/df9cb3849a06beb674e9d93160089fc84431260c